nixos/lib/systemd-unit-options.nix

```diff
@@ -284,71 +284,107 @@ in rec {
       '';
     };
 
-  }; };
+    script = mkOption {
+      type = types.lines;
+      default = "";
+      description = "Shell commands executed as the service's main process.";
+    };
 
-  stage2ServiceOptions = { name, config, ... }: {
-    imports = [
-      stage2CommonUnitOptions
-      serviceOptions
-    ];
+    scriptArgs = mkOption {
+      type = types.str;
+      default = "";
+      description = "Arguments passed to the main process script.";
+    };
 
-    options = {
-      script = mkOption {
-        type = types.lines;
-        default = "";
-        description = "Shell commands executed as the service's main process.";
-      };
+    preStart = mkOption {
+      type = types.lines;
+      default = "";
+      description = ''
+        Shell commands executed before the service's main process
+        is started.
+      '';
+    };
 
-      scriptArgs = mkOption {
-        type = types.str;
-        default = "";
-        description = "Arguments passed to the main process script.";
-      };
+    postStart = mkOption {
+      type = types.lines;
+      default = "";
+      description = ''
+        Shell commands executed after the service's main process
+        is started.
+      '';
+    };
 
-      preStart = mkOption {
-        type = types.lines;
-        default = "";
-        description = ''
-          Shell commands executed before the service's main process
-          is started.
-        '';
-      };
+    reload = mkOption {
+      type = types.lines;
+      default = "";
+      description = ''
+        Shell commands executed when the service's main process
+        is reloaded.
+      '';
+    };
 
-      postStart = mkOption {
-        type = types.lines;
-        default = "";
-        description = ''
-          Shell commands executed after the service's main process
-          is started.
-        '';
-      };
+    preStop = mkOption {
+      type = types.lines;
+      default = "";
+      description = ''
+        Shell commands executed to stop the service.
+      '';
+    };
 
-      reload = mkOption {
-        type = types.lines;
-        default = "";
-        description = ''
-          Shell commands executed when the service's main process
-          is reloaded.
-        '';
-      };
+    postStop = mkOption {
+      type = types.lines;
+      default = "";
+      description = ''
+        Shell commands executed after the service's main process
+        has exited.
+      '';
+    };
 
-      preStop = mkOption {
-        type = types.lines;
-        default = "";
-        description = ''
-          Shell commands executed to stop the service.
-        '';
-      };
+    jobScripts = mkOption {
+      type = with types; coercedTo path singleton (listOf path);
+      internal = true;
+      description = "A list of all job script derivations of this unit";
```

Comment:
Can you add to the description that this does not copy the closure? Just so that people are aware of the limitations